File: csharp/etl/ETL.cs
using System.Collections.Generic;
using System.Linq;

public class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        Dictionary<string, int> transformed = new Dictionary<string, int>();

        foreach (var element in old)
        {
            element.Value.ToList().ForEach(e => transformed[e.ToLower()] = element.Key);
        }

        return transformed;
    }
}

File: javascript/etl/etl.js
class ETL {
    constructor() {}

    transform(oldData) {
        let newData = {};

        for (let key in oldData) {
            oldData[key].forEach(e => newData[e.toLowerCase()] = +key);
        }
        
        return newData;
    }
};

module.exports = ETL;
